core/src/main/scala/kafka/api/TopicMetadata.scala

```diff
@@ -61,25 +62,25 @@ case class TopicMetadata(topic: String, partitionsMetadata: Seq[PartitionMetadat
     val topicMetadataInfo = new StringBuilder
     topicMetadataInfo.append("{TopicMetadata for topic %s -> ".format(topic))
     errorCode match {
-      case ErrorMapping.NoError =>
+      case e if e == Errors.NONE.code =>
```

Comment:
I'm not suggesting changing the response objects. In this case, you could just do 
```scala
Errors.forCode(errorCode) match {
  case Errors.NONE =>
  ...
}
```
There are a lot of local cases in this patch where you could do something like this to avoid all of the tedious `.code` calls.